.github/workflows/xcode.yml

```diff
@@ -0,0 +1,72 @@
+name: Xcode
+
+# build on c/cpp changes or workflow changes
+on:
+  push:
+    paths:
+      - '**/CMakeLists.txt'
+      - 'cmake/Modules/**'
+      - '.github/workflows/Xcode.yml'
+  pull_request:
+    paths:
+      - '**/CMakeLists.txt'
+      - 'cmake/Modules/**'
+      - '.github/workflows/Xcode.yml'
+
+jobs:
+  build:
+    runs-on: macos-14
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install deps
+        run: |
+          source ./util/ci/common.sh
+          install_macos_deps
+          # brew jsoncpp do not include libjsoncpp.a, and if installed header conflict caused build failure
+          brew uninstall jsoncpp
+
+      - name: Build with Cmake
+        run: |
+          mkdir build
+          cd build
+          cmake .. \
+            -DCMAKE_OSX_DEPLOYMENT_TARGET=14 \
+            -DCMAKE_FIND_FRAMEWORK=LAST \
+            -DCMAKE_INSTALL_PREFIX=../build/macos/ \
+            -DRUN_IN_PLACE=FALSE -DENABLE_GETTEXT=TRUE \
+            -DENABLE_SYSTEM_JSONCPP=OFF
+          cmake --build . -j$(sysctl -n hw.logicalcpu)
+          make install
+
+      - name: Build and Archive with Xcode
+        run: |
+          mkdir build_xcode
+          cd build_xcode
+          cmake .. \
+            -DCMAKE_FIND_FRAMEWORK=LAST \
+            -DRUN_IN_PLACE=FALSE -DENABLE_GETTEXT=TRUE \
+            -DFREETYPE_LIBRARY=/opt/homebrew/lib/libfreetype.a \
+            -DGETTEXT_INCLUDE_DIR=/path/to/include/dir \
+            -DGETTEXT_LIBRARY=/opt/homebrew/lib/libintl.a \
+            -DLUA_LIBRARY=/opt/homebrew/lib/libluajit-5.1.a \
+            -DOGG_LIBRARY=/opt/homebrew/lib/libogg.a \
+            -DVORBIS_LIBRARY=/opt/homebrew/lib/libvorbis.a \
+            -DVORBISFILE_LIBRARY=/opt/homebrew/lib/libvorbisfile.a \
+            -DZSTD_LIBRARY=/opt/homebrew/lib/libzstd.a \
+            -DGMP_LIBRARY=/opt/homebrew/lib/libgmp.a \
```

Comment:
can't use dynamic libs with xcode?

Reply:
Yes, we can. But with this, we do not have to care about signing dynamic libs.